test/useAsync.test.ts

```diff
@@ -0,0 +1,85 @@
+import { useAsync, UseAsyncReturn } from '../src';
+import { renderHook } from '@testing-library/react-hooks';
+
+interface StarwarsHero {
+  name: string;
+}
+
+type StarwarsHeroArgs = {
+  asyncFunction: () => Promise<StarwarsHero[]>;
+};
+
+export const generateMockResponseData = (amount: number = 5): StarwarsHero[] =>
+  [...Array(amount).keys()].map(n => ({
+    id: n + 1,
+    name: `Starwars Hero ${n + 1}`,
+  }));
+
+describe('useAync', () => {
+  const fakeResults = generateMockResponseData();
+
+  it('should have a useAsync hook', () => {
+    expect(useAsync).toBeDefined();
+  });
+
+  it('should resolve a successful request', async () => {
+    const onSuccess = jest.fn();
+    const onError = jest.fn();
+
+    const { result, waitForNextUpdate } = renderHook<
+      StarwarsHeroArgs,
+      UseAsyncReturn<StarwarsHero[]>
```

Comment:
not familiar with renderHook but isn't type inference enough to provide the types here?

Reply:
Absolutely correct, types removed.